### PR TITLE
Create contract host functions

### PR DESCRIPTION
**Pull request by sisuresh**
Adds the `create_contract` and `create_contract_using_parent_id ` host functions.

There's a simple test case in this PR for  `create_contract`, and I've verified that it works with the core `InvokeHostFunctionTests` (I'll open the core PR once this PR and https://github.com/stellar/stellar-core/pull/3461 are merged . `create_contract_using_parent_id` has not been tested yet.

### CODE REVIEW

File: stellar-contract-env-common/src/env.rs [hunk 1]
```diff
@@ -181,7 +183,7 @@ macro_rules! call_macro_with_all_host_functions {
 
             mod crypto "c" {
                 {"$_", fn compute_hash_sha256(x:Object) -> Object}
-                {"$_", fn verify_sig_ed25519(x:Object, k:Object, s:Object) -> RawVal}
+                {"$0", fn verify_sig_ed25519(x:Object, k:Object, s:Object) -> ()}
```
Comment (leighmcculloch) [inline stellar-contract-env-common/src/env.rs:186]: All host fns should have a return value, and then if they don't have a value to return can return the Void static type.

File: stellar-contract-env-common/src/env.rs [hunk 2]
```diff
@@ -167,6 +167,8 @@ macro_rules! call_macro_with_all_host_functions {
                 {"$C", fn binary_insert(x:Object, i:RawVal, v:RawVal) -> Object}
                 {"$D", fn binary_append(v1:Object, v2:Object) -> Object}
                 {"$E", fn binary_slice(v:Object, i:RawVal, l:RawVal) -> Object}
+                {"$F", fn create_contract(v: Object, salt: Object, key: Object, sig: Object) -> Object}
+                {"$G", fn create_contract_using_parent_id(v: Object, salt: Object) -> Object}
```
Comment (jayz22) [inline stellar-contract-env-common/src/env.rs:171]: These should probably belong to the ledger "l" module.

File: stellar-contract-env-common/src/env.rs (hunk 1 above)
Comment (sisuresh) [reply stellar-contract-env-common/src/env.rs]: Fixed (sorry I had to rebase).

File: stellar-contract-env-common/src/env.rs (hunk 2 above)
Comment (sisuresh) [reply stellar-contract-env-common/src/env.rs]: done

**Pull request merged**